app/src/main/kotlin/co/netguru/baby/monitor/client/feature/server/ServerViewModel.kt

```diff
@@ -87,16 +97,43 @@ class ServerViewModel @Inject constructor(
         super.onCleared()
     }
 
-    fun handleRtcServerConnectionState(connectionObservable: Observable<RtcServerConnectionState>) {
-        compositeDisposable += connectionObservable.subscribeOn(Schedulers.io())
-            .observeOn(AndroidSchedulers.mainThread())
+    fun handleRtcServerConnectionState(webRtcServiceBinder: WebRtcService.Binder) {
+        compositeDisposable += webRtcServiceBinder.getConnectionObservable()
+            .subscribeOn(schedulersProvider.io())
+            .observeOn(schedulersProvider.mainThread())
             .subscribeBy(
-                onNext = { mutableRtcConnectionStatus.postValue(it) },
+                onNext = {
+                    handleStreamState(it)
+                    mutableRtcConnectionStatus.postValue(it)
```

Comment:
You are observing it on mainThread, shouldn't you set value with setter `mutableRtcConnectionStatus.value = it`? 🤔

Reply:
Yeah the  .observeOn(schedulersProvider.mainThread()) is quite obsolete here if everything is being set using postValue which does it on mainThread anyway. I'll change it